capi/src/layout_editor_state.rs

```diff
@@ -0,0 +1,96 @@
+//! Represents the current state of the Layout Editor in order to visualize it properly.
+use crate::{c_char, output_str};
+use livesplit_core::{layout::editor::State as LayoutEditorState, settings::Value as SettingValue};
+
+/// type
+pub type OwnedLayoutEditorState = Box<LayoutEditorState>;
+/// type
+pub type NullableOwnedLayoutEditorState = Option<OwnedLayoutEditorState>;
+
+/// drop
+#[no_mangle]
+pub extern "C" fn LayoutEditorState_drop(this: OwnedLayoutEditorState) {
+    drop(this);
+}
+
+/// Returns the number of components in the layout.
+#[no_mangle]
+pub extern "C" fn LayoutEditorState_components_len(this: &LayoutEditorState) -> usize {
+    this.components.len()
+}
+
+/// Returns the name of the component at the specified index.
+#[no_mangle]
+pub extern "C" fn LayoutEditorState_components_text(
```

Comment:
I think I rather prefer the singular `component_text` here (as well for the others), as you query for a specific component.

Reply:
I'll adjust the naming of all of the functions meant to query a `Vec<_>` to use singular names.